lxd/container_post.go

```diff
@@ -203,12 +245,151 @@ func containerPost(d *Daemon, r *http.Request) Response {
 	return OperationResponse(op)
 }
 
-// Special case migrating a container backed by ceph across two cluster nodes.
-func containerPostClusteringMigrateWithCeph(d *Daemon, c container, oldName, newName, newNode string) Response {
-	if c != nil && c.IsRunning() {
-		return BadRequest(fmt.Errorf("Container is running"))
+// Move a non-ceph container to another cluster node.
+func containerPostClusteringMigrate(d *Daemon, c container, oldName, newName, newNode string) Response {
+	cert := d.endpoints.NetworkCert()
+
+	var sourceAddress string
+	var targetAddress string
+
+	// Save the original value of the "volatile.apply_template" config key,
+	// since we'll want to preserve it in the copied container.
+	origVolatileApplyTemplate := c.LocalConfig()["volatile.apply_template"]
+
+	err := d.cluster.Transaction(func(tx *db.ClusterTx) error {
+		var err error
+
+		sourceAddress, err = tx.NodeAddress()
+		if err != nil {
+			return errors.Wrap(err, "Failed to get local node address")
+		}
+
+		node, err := tx.NodeByName(newNode)
+		if err != nil {
+			return errors.Wrap(err, "Failed to get new node address")
+		}
+		targetAddress = node.Address
+
+		return nil
+	})
+	if err != nil {
+		return SmartError(err)
+	}
+
+	run := func(*operation) error {
+		// Connect to the source host, i.e. ourselves (the node the container is running on).
+		source, err := cluster.Connect(sourceAddress, cert, false)
+		if err != nil {
+			return errors.Wrap(err, "Failed to connect to local node")
```

Comment:
Failed to connect to source cluster member

Reply:
Fixed.